util/src/org/commcare/util/cli/CliCommand.java

```diff
@@ -0,0 +1,87 @@
+package org.commcare.util.cli;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.commcare.util.CommCareConfigEngine;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+public abstract class CliCommand {
+    protected final String commandName;
+    protected final String helpTextDescription;
+    protected final String positionalArgsHelpText;
+
+    protected String[] args;
+    protected Options options;
+    protected CommandLine cmd;
```

Comment:
Depending on if you use getters or not, many of these class fields can either become `private` or `public`.
Run the linter to see which